core/txpool/celo_validation.go

```diff
@@ -15,29 +16,24 @@ var NonWhitelistedFeeCurrencyError = errors.New("Fee currency given is not white
 // FeeCurrencyValidator validates currency whitelisted status at the specified
 // block number.
 type FeeCurrencyValidator interface {
```

Comment:
Looks like the only thing this interface does is provides the `Balance` function. Looks like it could be another struct's responsibility. `CeloBackend` potentially, since `Balance` through a chain of proxy functions calls its `GetBalanceERC20` in the end anyway. Then we could get rid of this interface and corresponding struct completely.